test_suite/tests/json.rs

```diff
@@ -55,30 +45,27 @@ fn test_unit_struct() {
 	struct UnitStruct;
 
 	assert_json_for_type::<UnitStruct>(json!({
-		"id": {
-			"custom.name": 1,
-			"custom.namespace": [2],
-			"custom.params": [],
-		},
-		"def": {
-			"tuple_struct.types": []
+		"composite": {
+			"name": 1,
+			"namespace": [2],
```

Comment:
I think we should merge `name` and `namespace` into a non-empty array of string table indices called `path`.